functions-and-operators/locking-functions.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Locking Functions
+summary: Learn about user-level locking functions in TiDB.
+---
+
+# Locking Functions
+
+TiDB supports most of the user-level [locking functions](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html) available in MySQL 5.7.
+
+## Supported functions
+
+| Name                                                                                                                 | Description                                                           |
+|:---------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------|
+| [`GET_LOCK(lockName, timeout)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_get-lock)    | Acquires an advisory lock. The `lockName` parameter must be NO longer than 64 characters.          |
```

Comment:
Is characters meant to mean *bytes* here or any utf character? I just tested the master branch and a name "ö" cannot be more than 32 characters (2 byte character in utf8mb4).

Reply:
```
tidb> select get_lock(repeat("ä",33),1);
ERROR 3057 (HY000): Incorrect user-level lock name 'äääääääääääääääääääääääääääääääää'.
```

Reply:
No, I think this is a bug. The check for length is checking based on byte length, but in MySQL the max is expected to be measured in char-length:

```sql
mysql [localhost:8028] {msandbox} (test) > select get_lock(repeat("ä",33),1);
+-----------------------------+
| get_lock(repeat("ä",33),1)  |
+-----------------------------+
|                           1 |
+-----------------------------+
1 row in set (0.00 sec)

mysql [localhost:8028] {msandbox} (test) > select get_lock(repeat("ä",65),1);
ERROR 3057 (42000): Incorrect user-level lock name 'äääääääääääääääääääääääääääääääääääääääääääääääääääääääääääääääää'.
```

Reply:
Reported as https://github.com/pingcap/tidb/issues/34639